File: JavaScript/Sliding Window/2130. Maximum Twin Sum of a Linked List.js
/* 2130. Maximum Twin Sum of a Linked List
Difficulty: Medium
Submission Link: https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length/submissions/1387473300/ */

var maxVowels = function (s, k) {

    let target = ['a', 'e', 'i', 'o', 'u'];
    let substr = [];
    let n = 0;
    let result = 0;
    for (let i = 0; i < k; i++) {
        if (target.includes(s[i])) {
            substr.push(true)
            n++
        } else {
            substr.push(false)
        }
    }
    result = n;
    for (let i = k; i < s.length; i++) {
        let first = substr.shift();
        substr.push(target.includes(s[i]));
        if (first) {
            n--;
        };
        if (target.includes(s[i])) {
            n++;
        };
        result = Math.max(result, n);
    }
    return result;
};

File: C#/Sliding Window/2130. Maximum Twin Sum of a Linked List.cs
/* 2130. Maximum Twin Sum of a Linked List
Difficulty: Medium
Submission Link: https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length/submissions/ */

public class Solution {
    bool checkVowel(char c){
        return c=='a'||c=='e'||c=='i'||c=='o'||c=='u';
    }
    public int MaxVowels(string s, int k) {
        int i=0;
        int j=0;
        int n=s.Length;
        int count =0;
        int max=0;
        while(j<n){
         if(checkVowel(s[j])) count++;
         if(j-i+1==k){
          max=Math.Max(max,count);
          if(checkVowel(s[i])) count--;
          i++;
         }
         j++;
        }
        return max;
    }
}